From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 26 Aug 2018 22:30:39 +0100
Subject: exec: Avoid ABI change in 4.9.116
Forwarded: not-needed

Commit b9dd13488acb "exec: avoid gcc-8 warning for get_task_comm"
changed get_task_comm() from an exported function into a macro.  To
avoid an ABI change, add the exported function back.  But leave the
macro in place so newly built modules get the benefit of it.

---
--- a/fs/exec.c
+++ b/fs/exec.c
@@ -1945,3 +1945,10 @@ COMPAT_SYSCALL_DEFINE5(execveat, int, fd
 				  argv, envp, flags);
 }
 #endif
+
+#undef get_task_comm
+char *get_task_comm(char *buf, struct task_struct *tsk)
+{
+	return __get_task_comm(buf, TASK_COMM_LEN, tsk);
+}
+EXPORT_SYMBOL_GPL(get_task_comm);
